nautobot/docs/plugins/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,247 @@
+        )
+    ]
+```
+
+## Remove Site/Region Related Fields from Migrated Data Models
+
+After the data migration is successful, we need to remove the `site`/`region` fields from your data model so that Nautobot will be able to remove the `Site` and `Region` models. Note that we need to remove those attributes in a separate migration file from the previous since they are two different operations. You can do that by simply removing the `site`/`region` attributes from your model class:
+
+```python
+# models.py
+class ExampleModel(OrganizationalModel):
+    # note that site field is gone
+    location = models.ForeignKey(to="dcim.Location", on_delete=models.PROTECT, blank=True, null=True)
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+...
+```
+
+After removing the `site` attribute, make the migration file by running `nautobot-server makemigrations [app_name] -n [migration_name]`, for example:
+
+```shell
+nautobot-server makemigrations example_app -n remove_site_field_from_example_model
+```
+
+The migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('example_app', '0008_migrate_example_model_data_from_site_to_location'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='examplemodel',
+            name='site',
+        ),
+    ]
+```
+
+!!! important
+    Before you apply this migration, you have to add a `run_before` attribute in this migration file to make sure that you remove `site`/`region` fields before `Site` and `Region` models themselves are removed. **Without it, your migration files might be out of order and your app will not start**.
+
+```python
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name after we reorder migration files
+        ("dcim", "0034_remove_region_and_site"),
+    ]
+```
+
+The final migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+    
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name after we reorder migration files
+        ("dcim", "0034_remove_region_and_site"),
+    ]
+    dependencies = [
+        ("example_app", "0008_migrate_example_model_data_from_site_to_location"),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name="examplemodel",
+            name="site",
+        ),
+    ]
+```
+
+Apply the migration file by running `nautobot-server migrate [app_name]`, for example:
```

Comment:
Is this applying all three of the migrations that were just authored in a single pass? I don't see this step earlier in the file.


What version of Nautobot itself needs to be running, and which of its migrations must and must not have already run for this migration to be successful? (Presumably the `example_app` migration will fail if `dcim.0034_remove_region_and_site` has already run, for example?)